bundles/org.openhab.binding.metofficedatahub/src/main/java/org/openhab/binding/metofficedatahub/internal/MetOfficeDataHubSiteApiHandler.java

```diff
@@ -0,0 +1,890 @@
+                newLatitude = String.valueOf(userLocation.getLatitude());
+                newLongitude = String.valueOf(userLocation.getLongitude());
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        getLocalizedText("site.error.no-user-location"));
+                return;
+            }
+        } else {
+            String[] coordinates = config.location.split(",");
+            if (coordinates.length != 2) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        getLocalizedText("site.error.config"));
+                return;
+            }
+            newLatitude = coordinates[0].trim();
+            newLongitude = coordinates[1].trim();
+        }
+
+        if (newLatitude.length() > 1) {
+            double trueVal;
+            try {
+                trueVal = Double.parseDouble(newLatitude);
+                if (trueVal > 85 || trueVal < -85) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            getLocalizedText("site.error.lat"));
+                }
+            } catch (final NumberFormatException | NullPointerException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        getLocalizedText("site.error.lat-type"));
+            }
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    getLocalizedText("site.error.lat-missing"));
+        }
+
+        if (newLongitude.length() > 1) {
+            double trueVal;
+            try {
+                trueVal = Double.parseDouble(newLongitude);
+                if (trueVal > 180 || trueVal < -180) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            getLocalizedText("site.error.long"));
+                }
+            } catch (final NumberFormatException | NullPointerException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        getLocalizedText("site.error.long-type"));
+            }
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    getLocalizedText("site.error.long-missing"));
+        }
+
+        // In case of errors - fail fast and set the device to be offline
+        if (getThing().getStatus().equals(ThingStatus.OFFLINE)) {
+            return;
+        }
+
+        latitude = newLatitude;
+        longitude = newLongitude;
+
+        /**
+         * Setup the initial device's status
+         */
+        scheduler.execute(() -> {
+            scheduler.execute(() -> {
+                updateStatus(ThingStatus.ONLINE);
+                checkDataRequired();
+                reconfigurePolling();
```

Comment:
On first glance at the code you look right - just tracking through to see if there's some funny scenario that's not clear, or I've made another silly mistake here, its just strange that I added that there.

Reply:
So had a look through I know last year when this was written I was very careful about the poll cycle counts, because of the limits for this service and checking the polling behaviours. I suspect this is in case people add other points after the polling sequences are started. As it's already soak tested for the last year, I'm a bit hesitant to change it just in case it introduces some bug its there to account for that I cant spot at the moment. Are we good to leave this as is, I think its a good catch bug just don't want to risk modifying this behaviour at the moment.

Reply:
I will double back on this when I have a new quota for the API, from what I saw today.

Reply:
See new review comments, this can certianly be removed once you reset the `requiresHourlyData` and daily to their default values.